arangod/VocBase/vocbase.cpp

```diff
@@ -1372,6 +1458,55 @@ arangodb::Result TRI_vocbase_t::dropCollection(DataSourceId cid,
   }
 }
 
+arangodb::Result TRI_vocbase_t::validateCollectionParameters(
+    arangodb::velocypack::Slice parameters) {
+  bool valid = parameters.isObject();
+  if (valid) {
+    // check that the name does not contain any strange characters
+    std::string name = VelocyPackHelper::getStringValue(
+        parameters, StaticStrings::DataSourceName, "");
+    bool isSystem = VelocyPackHelper::getBooleanValue(
+        parameters, StaticStrings::DataSourceSystem, false);
+    bool extendedNames =
+        server().getFeature<DatabaseFeature>().extendedNamesForCollections();
+    valid =
+        CollectionNameValidator::isAllowedName(isSystem, extendedNames, name);
+  }
+
+  if (!valid) {
+    return {TRI_ERROR_ARANGO_ILLEGAL_NAME};
```

Comment:
Can we add a Message here, including the violating name?
This will help users to fix it.

The use-case I have in mind is a customer trying to create a graph with many collections, where one violates the naming scheme. "ILLEGAL_NAME" will not be too helpful there.

Reply:
Added collection name to the message.